File: gachisa-backend/src/main/java/com/gachisa/order/dto/OrderResponse.java
package com.gachisa.order.dto;

import com.gachisa.order.entity.DeliveryStatus;
import com.gachisa.order.entity.Order;
import java.math.BigDecimal;
import java.time.LocalDateTime;

public record OrderResponse(
        Long orderId,
        String orderNumber,
        Long participationId,
        Long paymentId,
        Long groupBuyId,
        Long productId,
        String productName,
        String productImageUrl,
        int quantity,
        int basePrice,
        BigDecimal discountRate,
        int discountAmount,
        int amount,
        boolean deliveryAddressRegistered,
        DeliveryStatus deliveryStatus,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    public static OrderResponse from(Order order) {
        return new OrderResponse(
                order.getId(),
                order.getOrderNumber(),
                order.getParticipationId(),
                order.getPaymentId(),
                order.getGroupBuyId(),
                order.getProductId(),
                order.getProductName(),
                order.getProductImageUrl(),
                order.getQuantity(),
                order.getBasePrice(),
                order.getDiscountRate(),
                order.getDiscountAmount(),
                order.getAmount(),
                order.getAddress() != null,
                order.getDeliveryStatus(),
                order.getCreatedAt(),
                order.getUpdatedAt()
        );
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/order/dto/OrderResponse.kt
package com.gachisa.order.dto

import com.gachisa.order.entity.DeliveryStatus
import com.gachisa.order.entity.Order
import java.math.BigDecimal
import java.time.LocalDateTime

data class OrderResponse(
    val orderId: Long,
    val orderNumber: String,
    val participationId: Long,
    val paymentId: Long,
    val groupBuyId: Long,
    val productId: Long,
    val productName: String,
    val productImageUrl: String?,
    val quantity: Int,
    val basePrice: Int,
    val discountRate: BigDecimal,
    val discountAmount: Int,
    val amount: Int,
    val deliveryAddressRegistered: Boolean,
    val deliveryStatus: DeliveryStatus,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
) {
    companion object {
        fun from(order: Order): OrderResponse = OrderResponse(
            orderId = order.id!!,
            orderNumber = order.orderNumber,
            participationId = order.participationId,
            paymentId = order.paymentId,
            groupBuyId = order.groupBuyId,
            productId = order.productId,
            productName = order.productName,
            productImageUrl = order.productImageUrl,
            quantity = order.quantity,
            basePrice = order.basePrice,
            discountRate = order.discountRate,
            discountAmount = order.discountAmount,
            amount = order.amount,
            deliveryAddressRegistered = order.address != null,
            deliveryStatus = order.deliveryStatus,
            createdAt = order.createdAt,
            updatedAt = order.updatedAt,
        )
    }
}
